frontend/tests/Tests.elm

```diff
@@ -12,45 +12,48 @@ import Test.Html.Selector exposing (attribute, tag)
 import Util exposing (truncateContent)
 
 
-all : Test
-all =
-    describe "All"
-        [ chunkingAlgorithm
-        , truncationAlgorithm
-        , urlGuessing
-        ]
-
-
-urlCases : List ( String, String )
-urlCases =
-    [ ( "https://example.com", "Tämä on linkki https://example.com" )
-    , ( "http://www.example.com", "ja toinen linkki ilman http-alkua www.example.com" )
-    , ( "http://example.com/sisältö", "ja vielä yksi ilman www:tä, mutta kauttaviivalla example.com/sisältö" )
-    , ( "http://example.com/mitätahansa", "ja vielä taas, jossa sulkeva sulje, pilkku, tai piste ei kuulu linkkiin http://example.com/mitätahansa." )
-    , ( "http://example.com", "Myös alkava sulje pitäisi urlittaa (http://example.com)" )
-    ]
-
-
 urlGuessing : Test
 urlGuessing =
     describe "Url guessing algorithm" <|
-        List.map
-            (\( expectedUrl, str ) ->
-                test "has a single link" <|
-                    \() ->
-                        let
-                            elements =
-                                H.div [] (PlainTextFormat.view str)
-                                    |> Query.fromHtml
-                                    |> Query.findAll [ tag "a" ]
-                        in
-                        Expect.all
-                            [ Query.count (Expect.equal 1)
-                            , Query.first >> Query.has [ attribute "href" expectedUrl ]
-                            ]
-                            elements
-            )
-            urlCases
+        [ testUrlGuess
+            { testString = "Tämä on linkki https://example.com"
+            , expectedUrl = "https://example.com"
+            }
```

Comment:
Very minor thing: expectedUrl and testString could also be defined in the same (alphabetical) order as in testUrlGuess function.